source/wp-content/themes/wporg-developer-2023/src/search-filters/view.js

```diff
@@ -0,0 +1,41 @@
+const init = () => {
+	const allButton = document.querySelector( '#wp-block-wporg-search-filters-all' );
+	const container = document.querySelector( '.wp-block-wporg-search-filters' );
+
+	if ( ! allButton || ! container ) {
+		return;
+	}
+
+	const inputs = container.querySelectorAll( 'input[type="checkbox"]' );
+
+	const hasFilterChecked = () => [ ...inputs ].some( ( input ) => input.checked );
+
+	const updateAllButtonState = () => {
+		allButton.setAttribute( 'aria-pressed', ! hasFilterChecked() );
+	};
+
+	allButton.addEventListener( 'click', ( event ) => {
+		event.preventDefault();
+		inputs.forEach( ( input ) => {
+			input.checked = false;
+		} );
+		updateAllButtonState();
+	} );
+
+	container.addEventListener( 'change', () => {
+		updateAllButtonState();
+	} );
+
+	container.addEventListener( 'keydown', ( event ) => {
```

Comment:
If you switch the label/checkbox behavior, you shouldn't need this either.

Reply:
Removed